packages/block-library/src/paragraph/editor.scss

```diff
@@ -15,3 +15,28 @@
 		opacity: 0;
 	}
 }
+
+
+// Precise partial text selections.
+// @todo: For now, proof of concept. But code needs to be moved to the most appropriate place.
+// Furthermore, this code right now is likely going to affect numerous nested blocks, so
+// it will be important to scope it right!
+// To keep in mind, user-select is currently inherited to all nodes inside.
+// One solution: apply the user-select properties dynamically during selection.
+
+// Disable text selection on the main container.
+// This is necessary to ensure that we can direct the browser towards the content that's actually
+// meant to be selected.
+.block-editor-block-list__layout {
+	user-select: none;
+}
+
+// Re-enable it on Paragraphs.
+.block-editor-rich-text__editable {
```

Comment:
Could we generalise this to `[contenteditable]`?

Reply:
Very probably! And feel free to push directly to this branch, or make an entirely new one.